spec/support/http_helpers.rb

```diff
@@ -21,4 +24,18 @@ def mock_http_request(options = {})
     # Generate response
     { stub: stub, request: request, response: http.request(request) }
   end
+
+  def wait_http_server(server, delay)
+    delay.times do |i|
+      uri = URI(server + '/')
+      begin
+        res = Net::HTTP.get_response(uri)
+        return true if res.code == '200'
+      rescue StandardError => e
+        puts e if i >= 3 # display errors only when failing repeatedly
```

Comment:
Using `puts` to signal problems doesn't seem sufficient. What's the intent here?

If we'd like to raise a warning in the test suite that is non-breaking, then it should probably use the log function to do so. If this indicates a serious problem or faulty test, we probably should raise an error and fail deadly (even if it only happens after so many attempts.)

We do something very similar to this using `try_wait_until` in `synchronization_helpers.rb`: maybe the HTTP implementation should use this to drive its looping behavior instead.